### PR TITLE
data: document diacritics use in the different languages

### PR DESCRIPTION
**Pull request by kemenaran**
- Separate the diacritics table and the ascii table
- Document the diacritics of each language

### CODE REVIEW

**Comment by tobiasvl** [comment]
Should the diacritics file perhaps be named `ascii_to_diacritic.asm`, to be consistent with `ascii_to_tile.asm` and to make it clear that it maps an ASCII character to the corresponding diacritic which is harder to understand now that it's no longer directly beneath the ASCII-to-tile table?

Or maybe `diacritics.asm` could have a short explanatory text on the top which explains this. Maybe even `ascii_to_tile.asm` should have a short text (in the non-English versions) that explains how diacritics work, since otherwise it might be a bit confusing to see that the ASCII character Ä maps to the tile A, with no explanation on how the umlauts get there.

**Comment by tobiasvl** [comment]
Other counter-intuitive things that we could potentially document here:
* The diacritics tiles are found in https://github.com/zladx/LADX-Disassembly/tree/document-diacritics/tree/main/src/gfx/items/inventory_overworld_items.dmg, which is hard to find on your own
* The value 1 in the `DiacriticsTable` maps to tile `$C9`, while 2 maps to `$C8`, so the order is the opposite of what you might expect

**Comment by kemenaran** [comment]
Ah, yes, I'll definitely document those two shortcoming.

Also, I wasn't sure about the file naming, because I'm not sure the diacritics table is actually indexed by ascii code – is it really the case?

**Comment by tobiasvl** [comment]
> Also, I wasn't sure about the file naming, because I'm not sure the diacritics table is actually indexed by ascii code – is it really the case?

Yes.

This is easier to see with my annotations in #460. The ASCII value is put into `bc` [here](https://github.com/zladx/LADX-Disassembly/pull/460/files#diff-c4911341abbe29a19fd64e33beb16a83dc754d0de31a026877edafaa4c9c2953R166) before calling `ReadTileValueFromDiacriticsTable` (called `ReadDialogBankFromTable` before my PR, which is a misnomer), which loads the address to `DiacriticsTable` into `hl` [here](https://github.com/zladx/LADX-Disassembly/pull/460/files#diff-84e4db15124403048717b4b64a1d10dea27c7b756354aaecfe8d2aa406649540R780) before adding `bc` and `hl` together to ultimately get the entry in `DiacriticsTable` indexed by ASCII code and load that value into `a` before returning and using that value to decide what diacritic to use [here](https://github.com/zladx/LADX-Disassembly/pull/460/files#diff-c4911341abbe29a19fd64e33beb16a83dc754d0de31a026877edafaa4c9c2953R182).

File: revisions/J0/src/data/diacritics.asm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+; The Japanese text uses two diacritics:
+; - 1: ゛
```
Comment (tobiasvl) [inline revisions/J0/src/data/diacritics.asm:2]:
```suggestion
; - 1: ゛ (dakuten, tile $C9)
```

File: revisions/J0/src/data/diacritics.asm [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+; - 2: ゜
```
Comment (tobiasvl) [inline revisions/J0/src/data/diacritics.asm:3]:
```suggestion
; - 2: ゜ (handakuten, tile $C8)
```

File: revisions/F0/src/data/diacritics.asm [hunk 3]
```diff
@@ -0,0 +1,29 @@
+; The French text uses two diacritic tiles:
+; - 1: ´ (Acute accent on uppercase letter)
```
Comment (tobiasvl) [inline revisions/F0/src/data/diacritics.asm:2]:
```suggestion
; - 1: ´ (Acute accent on uppercase letter, tile $C9)
```

File: revisions/F0/src/data/diacritics.asm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+; - 2: ` (Grave accent on uppercase letter)
```
Comment (tobiasvl) [inline revisions/F0/src/data/diacritics.asm:3]:
```suggestion
; - 2: ` (Grave accent on uppercase letter, tile $C8)
```

File: revisions/G0/src/data/diacritics.asm [hunk 5]
```diff
@@ -0,0 +1,20 @@
+; The German text uses two diacritics:
+; - 1: ¨ (umlaut on uppercase letter)
```
Comment (tobiasvl) [inline revisions/G0/src/data/diacritics.asm:2]:
```suggestion
; - 1: ¨ (umlaut on uppercase letter, tile $C9)
```

File: revisions/G0/src/data/diacritics.asm [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+; - 2: ¨ (umlaut on lowercase letter)
```
Comment (tobiasvl) [inline revisions/G0/src/data/diacritics.asm:3]:
```suggestion
; - 2: ¨ (umlaut on lowercase letter, tile $C8)
```

File: revisions/J0/src/data/diacritics.asm [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+DiacriticsTable::
+    db     0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0
```
Comment (tobiasvl) [inline revisions/J0/src/data/diacritics.asm:5]: Maybe make the table a bit more compact with less whitespace, like in the other revisions?

**Review by tobiasvl — CHANGES_REQUESTED** [review]
Suggestions:

* Rename `diacritics.asm` to `ascii_to_diacritic.asm`, to be consistent with `ascii_to_tile.asm`
* Perhaps: Add some text to `ascii_to_tile.asm` for non-English revisions explaining where to find diacritics
* Not sure: Explain what tilesheet (https://github.com/zladx/LADX-Disassembly/tree/document-diacritics/tree/main/src/gfx/items/inventory_overworld_items.dmg) the two diacritics tiles are in

**Comment by kemenaran** [comment]
@tobiasvl I finally had some time to work again on this; it should be in better shape now. Thanks for your comments!

**Review by tobiasvl — APPROVED** [review]

**Pull request merged**